### ISSUE TITLE
"callback" argument must be a function

### ISSUE THREAD

**Issue by mktn87** [theturtle32/WebSocket-Node#266]
I'm getting the following error after starting the server:

```javascript
TypeError: "callback" argument must be a function
    at exports.setTimeout (timers.js:327:11)
    at WebSocketConnection.setKeepaliveTimer (pathToProject/node_modules/websocket/lib/WebSocketConnection.js:215:32)
    at WebSocketConnection.handleSocketData (pathToProject/node_modules/websocket/lib/WebSocketConnection.js:263:10)
    at emitOne (events.js:96:13)
    at TLSSocket.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:176:18)
    at TLSSocket.Readable.push (_stream_readable.js:134:10)
    at TLSWrap.onread (net.js:548:20)
```

I'm using version 6.9.1 of node.

**Comment by letsrock85** [comment]
I getting the same issue with promises functionality

**Comment by ibc** [comment]
> I'm getting the following error after starting the server

There is no info enough to consider this a bug. It's probably a wrong usage, but we don't know since you don't show any API usage.

> I getting the same issue with promises functionality

??

**Comment by Pauan** [comment]
@ibc I think it is a bug, please see pull request #251

I also got the same error.

I fixed it by changing `useNativeKeepalive` from `true` to `false` (that is the only change I made)

```
const wss = new WebSocketServer({
  httpServer: server,
  keepalive: true,
  useNativeKeepalive: false,
  autoAcceptConnections: false,
  disableNagleAlgorithm: false,
  ignoreXForwardedFor: true
});
```

**Comment by braytak** [comment]
I can confirm that the following combination of server options does cause the error to appear (as stated by @Pauan ); for web as well as for node clients:

``keepalive: true,    // default
useNativeKeepalive: true

Simply setting the native option to `false` makes the error go away.

**Comment by braytak** [comment]
Ooops! 
For forgot to specify that I am using node 6.9.2 and WebSocket 1.0.23.
My applications were running fine under node 5.9.0 and together with same version of WebSocket.

**Comment by braytak** [comment]
@ibc: a debug run revealed the following and a fix:

- `WebSocketConnection.handleSocketData`  calls the object's `setKeepaliveTimerHandler` which references the object's `_keepaliveTimerHandler` property as the `setTimeout` function
- since the `WebSocketConnection` constructor function creates the referenced property only in the case where the `keepalive` and `useNativeKeepalive` configuration members are set to `true' and `false` respectively, the `setTimeout` function argument is the undefined.

I got rid of the exception by moving the  `_keepaliveTimerHandler` property definition out of the conditional statement (to right after the `_closeTimerHandler` property definition).  

Cheers

**Comment by ibc** [comment]
Clear. Thanks a lot. Let continue on #251

**Comment by theturtle32** [comment]
Fixed by #251

**Comment by theturtle32** [comment]
Fix released as v1.0.24
